test cases/windows/17 gui app/meson.build

```diff
@@ -0,0 +1,22 @@
+project('gui_app_test', 'c')
+
+#
+# test that linking a Windows console applications with the main function in a
+# library is correctly instructed which entrypoint function to look for
+#
+
+console_lib = static_library('main', 'console_prog.c')
+executable('console', 'dummy.c', link_with: console_lib, gui_app: false)
+
+#
+# also verify that the correct subsystem is set by executable(gui_app:)
+#
+
+gui_prog = executable('gui_prog', 'gui_prog.c', gui_app: true)
+console_prog = executable('console_prog', 'console_prog.c', gui_app: false)
+
+tester = find_program('gui_app_tester.py')
+
+tool = find_program('objdump', 'dumpbin')
+test('is_gui', tester, args: [tool.path(), gui_prog, '2'])
```

Comment:
Will this test pop a GUI window during testing? It's not a problem, I'm just curious.

Reply:
No, it does the opposite. ``/SUBSYSTEM:WINDOWS`` is only a hint for application to prevent creation of a console window on launch. When launched via console, the console window is ignored and does not wait for the program to finish.

Reply:
Further, `gui_app_tester.py` doesn't run the built executable, it runs objdump on it and inspects the output.